core/runtime/impl/wasm_memory_impl.cpp

```diff
@@ -132,6 +132,13 @@ namespace kagome::runtime {
     return get<std::array<uint8_t, 16>>(addr);
   }
 
+  common::Buffer WasmMemoryImpl::loadN(kagome::runtime::WasmPointer addr,
+                                       kagome::runtime::SizeType n) const {
+    auto first = memory_.begin() + addr;
+    auto last = first + n;
```

Comment:
What if this value is out of bounds? The buffer will not contain exactly `n` elements. Maybe also a parameter for `written` should be returned?